x/params/module.go

```diff
@@ -0,0 +1,34 @@
+package params
+
+import (
+	"encoding/json"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+var (
+	_ sdk.AppModuleBasic = AppModuleBasic{}
```

Comment:
> It is (intentionally) only the basics are required (no "full" functionality is ever used by this module)

Would it be cleaner to set the `AppModule` functions to `nil` as we do with other modules ? I'm confused when to set the full `AppModule` vs only the `AppModuleBasic`.

A comment here on the code would be very helpful to clarify this.

Reply:
lol k I added in a code comment and then it seemed just way to extra... 

> Would it be cleaner to set the AppModule functions to nil as we do with other modules ?

lol no.

If _no_ functions from `AppModule` are ever defined anywhere in the module, then `AppModule` doesn't need to be defined, because it would be useless to add this type into the module manager. This is the case with params... none of the functions as defined by the `AppModule` interface are ever defined within the module, then we simply exclude it!

Reply:
I think this may be a bit clearer when we split up `AppModule` into `GenesisAppModule` and `AppModule` as chris suggested, then a few other modules will also not define `AppModule`